### ISSUE TITLE
class_weight based on predicted labels rather than true labels

### ISSUE THREAD

**Issue by jdevoldere** [tensorflow/tensorflow#40834]
**System information**
- TensorFlow version (you are using): tensorflow 2.0
- Are you willing to contribute it (Yes/No): No, I am currently unable to do so.

**Describe the feature and the current behavior/state.**

Let's stick to a two-class classification problem as an example to keep things simple so we can borrow concepts from the confusion matrix.

So I assume (since this doesn't seem to be documented) that class_weight weighs based on the true labels.
If this is the case then this would allow us to increase the recall of our model, as doing so puts more emphasis on true positives and false negatives.

But what if we want to increase precision instead? We should be able to do so by weighing classes based on the predicted labels instead, as this would put more emphasis on true positives and false positives instead. 

**Will this change the current api? How?**

This could be done by creating an extra parameter for the fit function called "predicted_class_weight".
Then to avoid confusion the old "class_weight" should be renamed to "true_class_weight".

**Who will benefit with this feature?**

One of the biggest drawbacks of the Keras API in its current state is that it's seemingly impossible to put emphasis on precision rather than recall.

There are numerous classification problems that require precision over recall. So this would benefit a lot of data scientists.

**Any Other info.**

/

**Comment by karmel** [comment]
Instead of adding a parameter, you can override the model train_step to add custom weighting there. Read more about overriding train_step here: https://keras.io/guides/customizing_what_happens_in_fit/
